src/components/molecules/NavSearchBar/NavSearchBar.tsx

```diff
@@ -1,69 +1,76 @@
-import React, { useCallback, useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState, ChangeEvent } from "react";
+import classNames from "classnames";
+
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
 import { VenueEvent } from "types/venues";
 import { Room, RoomTypes } from "types/rooms";
 import { User } from "types/User";
 
 import { WithId } from "utils/id";
-import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 import { isTruthy } from "utils/types";
+import { formatToWeekdayFullname, formatUtcSeconds } from "utils/time";
+import { currentVenueSelectorData, venueEventsSelector } from "utils/selectors";
 
 import { useWorldUsers } from "hooks/users";
 import { useSelector } from "hooks/useSelector";
 import { useProfileModalControls } from "hooks/useProfileModalControls";
 
+import { InputField } from "components/atoms/InputField";
 import { RoomModal } from "components/templates/PartyMap/components";
 
+import { NavSearchResult } from "./NavSearchResult";
+
 import "./NavSearchBar.scss";
-import { InputField } from "components/atoms/InputField";
-import { faSearch } from "@fortawesome/free-solid-svg-icons";
 
-interface SearchResult {
-  rooms: Room[];
-  users: readonly WithId<User>[];
-  events: VenueEvent[];
-}
+type WithImageUrl<T extends object> = T & { image_url?: string };
+
+const emptyEventsArray: VenueEvent[] = [];
 
 const NavSearchBar = () => {
   const [searchQuery, setSearchQuery] = useState("");
   const onSearchInputChange = useCallback(
-    (e) => setSearchQuery(e.target.value),
+    (e: ChangeEvent<HTMLInputElement>) => setSearchQuery(e.target.value),
     []
   );
 
-  const [searchResult, setSearchResult] = useState<SearchResult>({
-    rooms: [],
-    users: [],
-    events: [],
-  });
+  const [foundRooms, setFoundRooms] = useState<Room[]>([]);
+  const [foundUsers, setFoundUsers] = useState<readonly WithId<User>[]>([]);
+  const [foundEvents, setFoundEvents] = useState<WithImageUrl<VenueEvent>[]>(
+    []
+  );
```

Comment:
Why is only one of these marked as `readonly`? They probably all can be.

Reply:
marked all of them as `readonly`: https://github.com/sparkletown/sparkle/pull/1328/commits/1ae5d879a3e476a6ae049ff7058d7218029c4c1a